osx_utils.sh

```
@@ -288,7 +288,14 @@ function install_mac_cpython {
     local py_inst=$(pyinst_fname_for_version $py_version $py_osx_ver)
     local inst_path=$DOWNLOADS_SDIR/$py_inst
     mkdir -p $DOWNLOADS_SDIR
-    curl $MACPYTHON_URL/$py_stripped/${py_inst} > $inst_path
+    # catch 404 and other curl errors
+    set +e
```

Can you think of a way of not setting `-e` unconditionally here?   For example, by doing this in a function?

---

wrapped this in a utility function `cmd_notexit`, which captures the return code. It got a bit more complicated as `curl -f` seems to exit (rather than just returning non-zero) on failure in this travis environment, regardless of the status of `set -e` / `set +e`

---

Sorry to return to this when my mind is cluttered with other stuff, but can't you do something on the lines of:

```
curl -L https://no.site/index.html > bar || (retval=$? && if [ "$retval" == 6 ]; then echo "Could not resolve host"; fi)
```

?

---

sorry i missed this comment - answering for completeness. That wouldnt work when requesting a non-existent document from a valid website, which would happen with an invalid `MB_PYTHON_OSX_VER` or `MB_PYTHON_VERSION` (or combo). In that case the downloaded file contains a snippet of HTML with the http failure code (eg 404):
```
$ curl -L https://www.python.org/ftp/python/3.9.1/python-3.9.1-macosx10.88.pkg > bar.pkg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   146  100   146    0     0    802      0 --:--:-- --:--:-- --:--:--   802
$ echo $?
0
$ cat bar.pkg
<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```
This is the case if someone sets and invalid value of